onnxruntime/hosting/executor.cc

```diff
@@ -0,0 +1,118 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include <onnx/onnx_pb.h>
+#include "core/common/logging/logging.h"
+#include "core/framework/data_types.h"
+#include "core/framework/environment.h"
+#include "core/framework/framework_common.h"
+#include "core/framework/mem_buffer.h"
+#include "core/framework/ml_value.h"
+#include "core/framework/tensor.h"
+#include "core/framework/tensorprotoutils.h"
+
+#include "onnx-ml.pb.h"
+#include "predict.pb.h"
+
+#include "converter.h"
+#include "executor.h"
+
+namespace onnxruntime {
+namespace hosting {
+
+namespace protobufutil = google::protobuf::util;
+
+// TODO: make all logging has request id
+
+protobufutil::Status Executor::predict(const std::string& name, const std::string& version, const std::string& request_id,
+                                       onnxruntime::hosting::PredictRequest& request,
+                                       /* out */ onnxruntime::hosting::PredictResponse& response) {
+  bool using_raw_data = true;
+  auto logger = env_->GetLogger();
+
+  // Create the input NameMLValMap
+  onnxruntime::NameMLValMap nameMlValMap;
+  common::Status status;
+  for (const auto& input : request.inputs()) {
+    std::string input_name = input.first;
+    onnx::TensorProto input_tensor = input.second;
+    using_raw_data = using_raw_data && input_tensor.has_raw_data();
+
+    // Prepare the MLValue object
+    OrtAllocatorInfo* cpuAllocatorInfo;
+    auto ort_status = OrtCreateAllocatorInfo("Cpu", OrtDeviceAllocator, 0, OrtMemTypeDefault, &cpuAllocatorInfo);
+    if (ort_status != nullptr || cpuAllocatorInfo == nullptr) {
+      LOGS(logger, ERROR) << "OrtCreateAllocatorInfo FAILED! Input name: " << input_name;
+      return protobufutil::Status(protobufutil::error::Code::RESOURCE_EXHAUSTED, "OrtCreateAllocatorInfo() FAILED!");
+    }
+
+    size_t cpu_tensor_length;
+    status = onnxruntime::utils::GetSizeInBytesFromTensorProto<0>(input_tensor, &cpu_tensor_length);
+    if (!status.IsOK()) {
+      LOGS(logger, ERROR) << "GetSizeInBytesFromTensorProto() FAILED! Input name: " << input_name
+                          << ". Error code: " << status.Code()
+                          << ". Error Message: " << status.ErrorMessage();
+      return protobufutil::Status(static_cast<protobufutil::error::Code>(status.Code()), "GetSizeInBytesFromTensorProto() FAILED!");
+    }
+
+    std::unique_ptr<char[]> data(new char[cpu_tensor_length]);
```

Comment:
`std::vector`

Reply:
This is just a piece of memory we will eventually use below. std::vector or std::array seems adding unnecessary overhead here.